### ISSUE TITLE
Impossible to paint in game of life

### ISSUE THREAD

**Issue by NoodleSamaChan** [NoodleSamaChan/resume#5]
For the game of life game, the game doesn't seem to pick up the click of the mouse that are supposed to create the "cells". It is necessary to understand why and fix the issue so the game can be playable.

**Comment by NoodleSamaChan** [comment]
I investigated and the issue seems to be coming from here:


```rs
fn get_mouse_pos(&self, mouse: graphic::Mouse) -> Option<(f32, f32)> {
        let ret = self.input.pointer.interact_pos()?;
        let retured_pos = (ret[0], ret[1]);
        println!("mouse position: {:#?}", retured_pos);
        let clicked = match mouse {
            graphic::Mouse::Left => self.input.pointer.primary_clicked(),
            graphic::Mouse::Right => self.input.pointer.secondary_clicked(),
            graphic::Mouse::Discard => self.input.pointer.secondary_clicked(),
        };
        if !clicked {
            println!("i went into clicked");
            Some(retured_pos)
        } else {
            println!("no pos returned");
            None
        }
    }
```
When I printed to see what the positions looked like, I got this print back: 
```rs
mouse position: (
    585.41016,
    214.46094,
)
```
and I get a panic message when I wrote !clicked with this message:
```rs
Tried to index in a buffer of width 30 with a x of 586
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
this means that we never get a true from the match mouse. It appears that the mouse position is wildly different from the window buffer coordinates

**Comment by NoodleSamaChan** [comment]
I still investigated, and it appears the true issue comes from the fact that the matches in these 2 functions:
```rs
fn get_mouse_pos(&self, mouse: graphic::Mouse) -> Option<(f32, f32)> {
        let ret = self.input.pointer.interact_pos()?;
        let retured_pos = (ret[0], ret[1]);
        println!("mouse position: {:#?}", retured_pos);
        let clicked = match mouse {
            graphic::Mouse::Left => self.input.pointer.primary_clicked(),
            graphic::Mouse::Right => self.input.pointer.secondary_clicked(),
            graphic::Mouse::Discard => self.input.pointer.secondary_clicked(),
        };
        if !clicked {
            println!("i went into clicked");
            Some(retured_pos)
        } else {
            println!("no pos returned");
            None
        }
    }

    fn get_mouse_down(&self, mouse: graphic::Mouse) -> bool {
        println!("mouse clicked");
        match mouse {
            graphic::Mouse::Left => self.input.pointer.primary_pressed(),
            graphic::Mouse::Right => self.input.pointer.secondary_clicked(),
            graphic::Mouse::Discard => self.input.pointer.secondary_clicked(),
            
        }
    }
    ```
    always return false

**Comment by NoodleSamaChan** [comment]
with #189d9bc, I fixed the fact that the bool were always false. They are now true when mouse is clicked.

However, there is still the issue with the buffer and the mouse position that don't appear to match

**Comment by NoodleSamaChan** [comment]
fixed by #25ffbf2
